src/page/product-list.page.ts

```diff
@@ -1,16 +1,30 @@
 import {
-  $, ElementFinder, ExpectedConditions, browser,
+  $, $$, ElementArrayFinder, ElementFinder,
+  ExpectedConditions, browser,
 } from 'protractor';
 
 export class ProductListPage {
-  private addToCardMenu: ElementFinder;
+  private addToCardBttn: ElementFinder;
+  private products: ElementArrayFinder;
 
   constructor() {
-    this.addToCardMenu = $('#center_column [title="Add to cart"]');
+    this.products = $$('.product_list.row');
+    this.addToCardBttn = $('#add_to_cart > button');
+
+  }
+
+  private findByProduct(productName: string): ElementFinder {
+    return this.products.filter((product: ElementFinder) => product.$('[itemprop="name"]').getText().then((text : string) => text === productName)).first();
+  }
+
+  public async selectProduct (productName: string): Promise<void> {
+    let product = this.findByProduct(productName);
```

Comment:
let´s use `const` instead of `let` since the value of the `product` should not change.

Reply:
https://medium.com/@tatymolys/var-let-y-const-donde-cuando-y-por-qu%C3%A9-d4a0ee66883b